src/components/jobs/create/jobCreateDetailsDialog.ts

```diff
@@ -140,7 +136,29 @@ export class JobCreateDetailsDialog {
         })
 
         this.job.target_location = this.job.source_location;
-        this.job.name = this.vm_name + '-' + selected_dataset.name;
+        
+        const virtual_machine = _.find(this.virtual_machines, virtual_machine => {
+            return this.job.sdc_vm_id === virtual_machine.id;
+        });
+
+        if (virtual_machine.type === 'kvm') {
```

Comment:
This probably should be pulled out into a function to get the name. Function could/should take in vmType, datasetType, datasetName and return the correct name. This will make for much easier unit testing.